### DISCUSSION TITLE
Disabling Livewire DevTools in Production Environment

### DISCUSSION THREAD

**Discussion by Bagmaz** [livewire/livewire#8633] [1. Help] [answered]
Hi,

I am using Laravel 11 and Livewire 3, and I have Livewire DevTools installed in Chrome. When my application is set to `APP_ENV=production` and `APP_DEBUG=false`, the DevTools are still visible and editable.

What steps should I take to disable Livewire DevTools in a production environment?

<img width="191" alt="Screenshot 2024-07-07 at 09 04 33" src="https://github.com/livewire/livewire/assets/20259560/d783cb4e-cd19-49b7-bf0f-9cfc3828cd02">
<img width="1373" alt="Screenshot 2024-07-07 at 09 05 36" src="https://github.com/livewire/livewire/assets/20259560/f74d03dd-97e9-4dee-9de2-93f8edb408a6">

Thank you!

**Comment by joshhanley** [comment]
@Bagmaz devtools isn't maintained by us, so you would need to contact whoever built it.

**Reply by Bagmaz** [reply]
Thank you for getting back to me.

I understand that the Livewire DevTools are not managed by the Livewire team. But I need some help figuring out why the DevTools are still showing up in my app, even though it's set to APP_ENV=production and APP_DEBUG=false.

Can you explain how third-party tools like Livewire DevTools can access Livewire? I’m curious about how these tools can still interact with Livewire in a production setup.

Also, what can I do to make sure Livewire DevTools (or similar tools) are not visible and editable in production? Are there any specific settings or code changes I need to make to turn them off in a live environment?

Any advice you can offer would be really helpful. Thanks a lot!

**Reply by PhiloNL** [reply, marked as answer]
@Bagmaz the devtools use public information that is accessible via the global `Livewire` javascript object. This does not change in production or if debugging is disabled. Any **public properties** in your component are shared with the front end. If you don't need properties to be public, I recommend using computed properties.
